dbuf-lsp/src/core/errors/rename_errors.rs

```diff
@@ -4,91 +4,77 @@ use tower_lsp::jsonrpc::Error;
 use tower_lsp::jsonrpc::ErrorCode::ServerError;
 use tower_lsp::jsonrpc::Result;
 
-/// Returns param is incorrect error.
-fn bad_rename_error(text: &'static str, code: i64) -> Error {
+use strum_macros::EnumIter;
+
+#[derive(EnumIter)]
+pub enum RenameError {
+    ToEmpty,
+    ToBuiltin,
+    ToKeyword,
+    OfBuiltin,
+    ToPrevious,
+    OfNone,
+    ToBadType(String),
```

Comment:
Since `strum`'s macros wouldn't help here, you can take a look at [`thiserror`](https://crates.io/crates/thiserror) for more ergonomic deriving of error descriptions.